flink-runtime/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/CommittableCollector.java

```diff
@@ -254,4 +209,31 @@ private CheckpointCommittableManagerImpl<CommT> getCheckpointCommittables(
                 this.checkpointCommittables.get(committable.getCheckpointIdOrEOI());
         return checkNotNull(committables, "Unknown checkpoint for %s", committable);
     }
+
+    /** Removes all metadata about checkpoints of which all committables are fully committed. */
+    public void compact() {
```

Comment:
Nit: This change looks unrelated to the overall change of this PR. I am okay with removing the side-effect of cleaning the committables on `get`.

Reply:
Fair point that it's a bit mixed but it was also not big enough to be a commit on its own imho. So I'd leave as-is.